application/models/post.go

```diff
@@ -745,3 +823,59 @@ func (p *Post) GetLocationForNotifications() (*Location, error) {
 	}
 	return &postLocation, nil
 }
+
+// createNewHistory checks if the post has a status that is different than the
+// most recent of its Post History entries.  If so, it creates a new Post History
+// with the Post's new status.
+func (p *Post) createNewHistory() error {
```

Comment:
Should this be a `PostHistory` function: `func (p *PostHistory) New(post Post) error`?

Reply:
That seems less intuitive to me, since you don't start with a PostHistory that you're doing something with. Rather, you have a Post for which you want to create a new related PostHistory.

Reply:
I guess you could look at it that way. My thought was that almost every line in the function is operating on a PostHistory. Also, Post is becoming a "god class" and I think we need to be on the lookout for things that don't belong there.

Reply:
I'll want to get this merged for Billy's sake.  I'll make that change in the next one.